### ISSUE TITLE
Update process fails and loops

### ISSUE THREAD

**Issue by marek1712** [ansible/tower-operator#1412]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx-operator/issues) for duplicates.
- [X] I understand that the AWX Operator is open source software provided for free and that I might not receive a timely response.

### Bug Summary

I tried updating AWX 21.14.0 to v22 (22.0.0, 22.1.0, 22.2.0) but end up with with same results.
I tried making something out of the attached logs but honestly, I'm not sure what should I be looking for.
I doubt this is the reason:

### AWX Operator version

2.1.0

### AWX version

22.1.0

### Kubernetes platform

kubernetes

### Kubernetes/Platform version

k3s

### Modifications

no

### Steps to reproduce

cd ~
git clone https://github.com/ansible/awx-operator.git
cd awx-operator
git checkout 2.1.0

export NAMESPACE=awx
make deploy

Then watch the update process. At some point process loops:
kubectl -n awx logs -f deployments/awx-operator-controller-manager -c awx-manager

### Expected results

AWX updated to 22.1.0

### Actual results

Update process fails and loops

### Additional information

I followed this guide to deploy AWX/AWX-Operator in Rancher:
https://github.com/kurokobo/awx-on-k3s/

### Operator Logs

[logs.txt](https://github.com/ansible/awx-operator/files/11487432/logs.txt)

**Comment by kurokobo** [comment]
@marek1712 
Hi, thanks for using my guide.
From the log file, you can see the task `Apply deployment resources` has been failed. Upon further inspection, the deployment of `awx-task` has been failed with `Timed out waiting on resource`.

You should investigate the `STATUS` of the `awx-task` pod, `Events` for the pod, and logs from it.

```bash
# Get STATUS, it should be Running
$ kubectl -n awx get pod
NAME                                              READY   STATUS    RESTARTS   AGE
...
awx-task-**********-*****                         4/4     Running   0          6d16h
...

# Gather Events field of the awx-task pod
$ kubectl -n awx describe pod awx-task-**********-*****
...
Events:
  Type     Reason            Age   From               Message
  ----     ------            ----  ----               -------
  ...

# Gather logs from awx-task pod
$ kubectl -n awx logs -f deployment/awx-task -c init
$ kubectl -n awx logs -f deployment/awx-task -c init-projects
$ kubectl -n awx logs -f deployment/awx-task -c awx-task
$ kubectl -n awx logs -f deployment/awx-task -c awx-ee
$ kubectl -n awx logs -f deployment/awx-task -c awx-rsyslog
$ kubectl -n awx logs -f deployment/awx-task -c redis
```

I think your `awx-task` pod is in either `Pending`, `Crashloopbackoff`, or `Init:CrashLoopBackOff`. In any case, the results of the above commands would help.

**Comment by marek1712** [comment]
Hi @kurokobo 
Thank you for providing that amazing guide!

Indeed, awx-task is stuck in ```Crashloopbackoff```.
Please see the logs:

```
[root@awx ~]# kubectl -n awx get pod
NAME                                               READY   STATUS             RESTARTS         AGE
awx-web-7bbbdb9c68-psmbn                           3/3     Running            33 (9m10s ago)   41d
awx-postgres-13-0                                  1/1     Running            26 (9m10s ago)   196d
awx-7559596f7c-zkjv9                               4/4     Running            4 (9m10s ago)    40d
awx-operator-controller-manager-6b57459f99-qvwkv   2/2     Running            2 (9m10s ago)    40d
awx-task-78b5875955-w2894                          3/4     CrashLoopBackOff   92 (2m59s ago)   41d
```

```
kubectl -n awx describe pod awx-task-78b5875955-w2894
Name:             awx-task-78b5875955-w2894
Namespace:        awx
Priority:         0
Service Account:  awx
Node:             awx.abc.local/192.168.233.128
Start Time:       Thu, 06 Apr 2023 12:00:31 +0200
Labels:           app.kubernetes.io/component=awx
                  app.kubernetes.io/managed-by=awx-operator
                  app.kubernetes.io/name=awx-task
                  app.kubernetes.io/operator-version=2.0.0
                  app.kubernetes.io/part-of=awx
                  app.kubernetes.io/version=22.0.0
                  pod-template-hash=78b5875955
Annotations:      checksum-configmaps-config: bd935dc89c055dbbd8f2844f8e06db25b11a4f09
                  checksum-configmaps-pre_stop_scripts: adc83b19e793491b1c6ea0fd8b46cd9f32e592fc
                  checksum-secret-bundle_cacert: da39a3ee5e6b4b0d3255bfef95601890afd80709
                  checksum-secret-ldap_cacert: da39a3ee5e6b4b0d3255bfef95601890afd80709
                  checksum-secret-receptor_ca: 7112f7d383fcd348270d17ae85aae9ae5368aa68
                  checksum-secret-receptor_work_signing: 0dfdeea8b24a7254f02fb56fe0d6db41b15caca6
                  checksum-secret-route_tls: da39a3ee5e6b4b0d3255bfef95601890afd80709
                  checksum-secret-secret_key: 74ed28c0cff87d66b2b34a8ac215214246af4eb2
                  checksum-secrets-app_credentials: b47046870a467b4ffcdebb166237615719c89e8a
                  checksum-storage-persistent: adc83b19e793491b1c6ea0fd8b46cd9f32e592fc
Status:           Running
IP:               10.42.0.127
IPs:
  IP:           10.42.0.127
Controlled By:  ReplicaSet/awx-task-78b5875955
Init Containers:
  init:
    Container ID:  containerd://f12afb6556e5a13a07be75bced5c22ac390da1da8bcddec1ba3e70f999854fe9
    Image:         quay.io/ansible/awx-ee:latest
    Image ID:      quay.io/ansible/awx-ee@sha256:8290dfe406505505bffe0af5950b82b550a1b2a75da74ae60a463775ecd0e345
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/sh
      -c
      hostname=$MY_POD_NAME
      receptor --cert-makereq bits=2048 commonname=$hostname dnsname=$hostname nodeid=$hostname outreq=/etc/receptor/tls/receptor.req outkey=/etc/receptor/tls/receptor.key
      receptor --cert-signreq req=/etc/receptor/tls/receptor.req cacert=/etc/receptor/tls/ca/receptor-ca.crt cakey=/etc/receptor/tls/ca/receptor-ca.key outcert=/etc/receptor/tls/receptor.crt verify=yes

    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Wed, 17 May 2023 20:16:00 +0200
      Finished:     Wed, 17 May 2023 20:16:00 +0200
    Ready:          True
    Restart Count:  3
    Environment:
      MY_POD_NAME:  awx-task-78b5875955-w2894 (v1:metadata.name)
    Mounts:
      /etc/receptor/tls/ from awx-receptor-tls (rw)
      /etc/receptor/tls/ca/receptor-ca.crt from awx-receptor-ca (ro,path="tls.crt")
      /etc/receptor/tls/ca/receptor-ca.key from awx-receptor-ca (ro,path="tls.key")
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-6dx5k (ro)
  init-projects:
    Container ID:  containerd://e7e5d700b8e0282f3424058fe27ee2f6a888d31edced43793106b236455dc6d5
    Image:         quay.io/centos/centos:stream9
    Image ID:      quay.io/centos/centos@sha256:2d3be96ca5098ec06fbb1e9fed64f257c6a3405687978618e2faa6c108fc028d
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/sh
      -c
      chmod 775 /var/lib/awx/projects
      chgrp 1000 /var/lib/awx/projects

    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Wed, 17 May 2023 20:16:01 +0200
      Finished:     Wed, 17 May 2023 20:16:01 +0200
    Ready:          True
    Restart Count:  0
    Environment:
      MY_POD_NAME:  awx-task-78b5875955-w2894 (v1:metadata.name)
    Mounts:
      /var/lib/awx/projects from awx-projects (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-6dx5k (ro)
Containers:
  redis:
    Container ID:  containerd://7f46e3c3c313a5633fa6d5422e4a5bf5a828d235e849d40804eed2960be07f83
    Image:         docker.io/redis:7
    Image ID:      docker.io/library/redis@sha256:ed8cba11c09451dbb3495f15951e4afb4f1ba72a4a13e135c6da06c6346e0333
    Port:          <none>
    Host Port:     <none>
    Args:
      redis-server
      /etc/redis.conf
    State:          Running
      Started:      Wed, 17 May 2023 20:16:02 +0200
    Last State:     Terminated
      Reason:       Unknown
      Exit Code:    255
      Started:      Fri, 07 Apr 2023 09:21:41 +0200
      Finished:     Wed, 17 May 2023 20:15:52 +0200
    Ready:          True
    Restart Count:  3
    Requests:
      cpu:        50m
      memory:     64Mi
    Environment:  <none>
    Mounts:
      /data from awx-redis-data (rw)
      /etc/redis.conf from awx-redis-config (ro,path="redis.conf")
      /var/run/redis from awx-redis-socket (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-6dx5k (ro)
  awx-task:
    Container ID:  containerd://7cfc3997e86c7e356c2fb24b9359b5041d6d6e1d44f2cde8d73eea75b63fff04
    Image:         quay.io/ansible/awx:22.0.0
    Image ID:      quay.io/ansible/awx@sha256:085c82ba8a5687c8755eb5349e5e41349496a29a6d831d276461672658ca860a
    Port:          <none>
    Host Port:     <none>
    Args:
      /usr/bin/launch_awx_task.sh
    State:          Running
      Started:      Wed, 17 May 2023 20:16:02 +0200
    Last State:     Terminated
      Reason:       Unknown
      Exit Code:    255
      Started:      Fri, 07 Apr 2023 09:48:30 +0200
      Finished:     Wed, 17 May 2023 20:15:52 +0200
    Ready:          True
    Restart Count:  16
    Environment:
      AWX_COMPONENT:           task
      SUPERVISOR_CONFIG_PATH:  /etc/supervisord_task.conf
      AWX_SKIP_MIGRATIONS:     1
      MY_POD_UID:               (v1:metadata.uid)
      MY_POD_IP:                (v1:status.podIP)
      MY_POD_NAMESPACE:        awx (v1:metadata.namespace)
    Mounts:
      /etc/receptor/ from awx-receptor-config (rw)
      /etc/receptor/signing/work-private-key.pem from awx-receptor-work-signing (ro,path="work-private-key.pem")
      /etc/tower/SECRET_KEY from awx-secret-key (ro,path="SECRET_KEY")
      /etc/tower/conf.d/credentials.py from awx-application-credentials (ro,path="credentials.py")
      /etc/tower/conf.d/execution_environments.py from awx-application-credentials (ro,path="execution_environments.py")
      /etc/tower/conf.d/ldap.py from awx-application-credentials (ro,path="ldap.py")
      /etc/tower/settings.py from awx-settings (ro,path="settings.py")
      /var/lib/awx/projects from awx-projects (rw)
      /var/run/awx-rsyslog from rsyslog-socket (rw)
      /var/run/receptor from receptor-socket (rw)
      /var/run/redis from awx-redis-socket (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-6dx5k (ro)
  awx-ee:
    Container ID:  containerd://4770f64af0aac65f7cc851577f806e91d9e06074eccc0174f15ab61b123fe3ba
    Image:         deployablenl/awx-controlplane-ee:latest
    Image ID:      docker.io/deployablenl/awx-controlplane-ee@sha256:ae8ff36470441841e9877664703adebac8010437a6fe018a5417252faf0dbc7a
    Port:          <none>
    Host Port:     <none>
    Args:
      /bin/sh
      -c
      if [ ! -f /etc/receptor/receptor.conf ]; then
        cp /etc/receptor/receptor-default.conf /etc/receptor/receptor.conf
        sed -i "s/HOSTNAME/$HOSTNAME/g" /etc/receptor/receptor.conf
      fi
      exec receptor --config /etc/receptor/receptor.conf

    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Wed, 17 May 2023 20:27:12 +0200
      Finished:     Wed, 17 May 2023 20:27:12 +0200
    Ready:          False
    Restart Count:  58
    Environment:    <none>
    Mounts:
      /etc/receptor/ from awx-receptor-config (rw)
      /etc/receptor/receptor-default.conf from awx-default-receptor-config (rw,path="receptor.conf")
      /etc/receptor/signing/work-private-key.pem from awx-receptor-work-signing (ro,path="work-private-key.pem")
      /etc/receptor/tls/ from awx-receptor-tls (rw)
      /etc/receptor/tls/ca/receptor-ca.crt from awx-receptor-ca (ro,path="tls.crt")
      /var/lib/awx/projects from awx-projects (rw)
      /var/run/receptor from receptor-socket (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-6dx5k (ro)
  awx-rsyslog:
    Container ID:  containerd://244107241fdd3e456c8d0ea1f67b9a52c70d2486d7a78962d4ac705fcbf43f44
    Image:         quay.io/ansible/awx:22.0.0
    Image ID:      quay.io/ansible/awx@sha256:085c82ba8a5687c8755eb5349e5e41349496a29a6d831d276461672658ca860a
    Port:          <none>
    Host Port:     <none>
    Args:
      /usr/bin/launch_awx_rsyslog.sh
    State:          Running
      Started:      Wed, 17 May 2023 20:16:03 +0200
    Last State:     Terminated
      Reason:       Unknown
      Exit Code:    255
      Started:      Fri, 07 Apr 2023 09:48:40 +0200
      Finished:     Wed, 17 May 2023 20:15:52 +0200
    Ready:          True
    Restart Count:  16
    Environment:
      SUPERVISOR_CONFIG_PATH:  /etc/supervisor_rsyslog.conf
    Mounts:
      /etc/tower/SECRET_KEY from awx-secret-key (ro,path="SECRET_KEY")
      /etc/tower/conf.d/credentials.py from awx-application-credentials (ro,path="credentials.py")
      /etc/tower/settings.py from awx-settings (ro,path="settings.py")
      /var/run/awx-rsyslog from rsyslog-socket (rw)
      /var/run/redis from awx-redis-socket (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-6dx5k (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  awx-application-credentials:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  awx-app-credentials
    Optional:    false
  awx-receptor-tls:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  awx-receptor-ca:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  awx-receptor-ca
    Optional:    false
  awx-receptor-work-signing:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  awx-receptor-work-signing
    Optional:    false
  awx-secret-key:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  awx-secret-key
    Optional:    false
  awx-settings:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      awx-awx-configmap
    Optional:  false
  awx-nginx-conf:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      awx-awx-configmap
    Optional:  false
  awx-redis-config:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      awx-awx-configmap
    Optional:  false
  awx-redis-socket:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  awx-redis-data:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  rsyslog-socket:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  receptor-socket:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  awx-receptor-config:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  awx-default-receptor-config:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      awx-awx-configmap
    Optional:  false
  awx-projects:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  awx-projects-claim
    ReadOnly:   false
  kube-api-access-6dx5k:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason          Age                  From     Message
  ----     ------          ----                 ----     -------
  Normal   Pulled          41d (x7 over 41d)    kubelet  Container image "quay.io/ansible/awx:22.0.0" already present on machine
  Warning  BackOff         41d (x506 over 41d)  kubelet  Back-off restarting failed container
  Normal   SandboxChanged  40d                  kubelet  Pod sandbox changed, it will be killed and re-created.
  Normal   Pulled          40d                  kubelet  Container image "quay.io/ansible/awx-ee:latest" already present on machine
  Normal   Created         40d                  kubelet  Created container init
  Normal   Started         40d                  kubelet  Started container init
  Normal   Pulled          40d                  kubelet  Container image "quay.io/centos/centos:stream9" already present on machine
  Normal   Created         40d                  kubelet  Created container init-projects
  Normal   Started         40d                  kubelet  Started container init-projects
  Normal   Pulled          40d                  kubelet  Container image "docker.io/redis:7" already present on machine
  Normal   Created         40d                  kubelet  Created container redis
  Normal   Started         40d                  kubelet  Started container redis
  Normal   Pulled          40d                  kubelet  Container image "quay.io/ansible/awx:22.0.0" already present on machine
  Normal   Created         40d                  kubelet  Created container awx-task
  Normal   Started         40d                  kubelet  Started container awx-task
  Normal   Pulled          40d                  kubelet  Container image "quay.io/ansible/awx:22.0.0" already present on machine
  Normal   Created         40d                  kubelet  Created container awx-rsyslog
  Normal   Started         40d                  kubelet  Started container awx-rsyslog
  Normal   Pulled          40d (x2 over 40d)    kubelet  Container image "deployablenl/awx-controlplane-ee:latest" already present on machine
  Normal   Created         40d (x2 over 40d)    kubelet  Created container awx-ee
  Normal   Started         40d (x2 over 40d)    kubelet  Started container awx-ee
  Warning  BackOff         40d (x150 over 40d)  kubelet  Back-off restarting failed container
  Warning  FailedMount     12m                  kubelet  MountVolume.SetUp failed for volume "awx-default-receptor-config" : failed to sync configmap cache: timed out waiting for the condition
  Warning  FailedMount     12m                  kubelet  MountVolume.SetUp failed for volume "awx-settings" : failed to sync configmap cache: timed out waiting for the condition
  Warning  FailedMount     12m                  kubelet  MountVolume.SetUp failed for volume "awx-redis-config" : failed to sync configmap cache: timed out waiting for the condition
  Normal   SandboxChanged  12m                  kubelet  Pod sandbox changed, it will be killed and re-created.
  Normal   Pulled          11m                  kubelet  Container image "quay.io/ansible/awx-ee:latest" already present on machine
  Normal   Created         11m                  kubelet  Created container init
  Normal   Started         11m                  kubelet  Started container init
  Normal   Pulled          11m                  kubelet  Container image "quay.io/centos/centos:stream9" already present on machine
  Normal   Created         11m                  kubelet  Created container init-projects
  Normal   Started         11m                  kubelet  Started container init-projects
  Normal   Pulled          11m                  kubelet  Container image "docker.io/redis:7" already present on machine
  Normal   Created         11m                  kubelet  Created container redis
  Normal   Started         11m                  kubelet  Started container redis
  Normal   Pulled          11m                  kubelet  Container image "quay.io/ansible/awx:22.0.0" already present on machine
  Normal   Created         11m                  kubelet  Created container awx-task
  Normal   Started         11m                  kubelet  Started container awx-task
  Normal   Created         11m                  kubelet  Created container awx-ee
  Normal   Started         11m                  kubelet  Started container awx-ee
  Normal   Pulled          11m                  kubelet  Container image "quay.io/ansible/awx:22.0.0" already present on machine
  Normal   Created         11m                  kubelet  Created container awx-rsyslog
  Normal   Started         11m                  kubelet  Started container awx-rsyslog
  Normal   Pulled          11m (x2 over 11m)    kubelet  Container image "deployablenl/awx-controlplane-ee:latest" already present on machine
  Warning  BackOff         113s (x47 over 11m)  kubelet  Back-off restarting failed container
```

```
kubectl -n awx logs -f deployment/awx-task -c awx-task
[wait-for-migrations] Waiting for database migrations...
[wait-for-migrations] Attempt 1 of 30
[wait-for-migrations] Waiting 0.5 seconds before next attempt
[wait-for-migrations] Attempt 2 of 30
[wait-for-migrations] Waiting 1 seconds before next attempt
[wait-for-migrations] Attempt 3 of 30
[wait-for-migrations] Waiting 2 seconds before next attempt
[wait-for-migrations] Attempt 4 of 30
[wait-for-migrations] Waiting 4 seconds before next attempt
[wait-for-migrations] Attempt 5 of 30
[wait-for-migrations] Waiting 8 seconds before next attempt
[wait-for-migrations] Attempt 6 of 30
[wait-for-migrations] Waiting 16 seconds before next attempt
[wait-for-migrations] Attempt 7 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 8 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 9 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 10 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 11 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 12 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 13 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 14 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 15 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 16 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 17 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 18 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 19 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 20 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 21 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 22 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 23 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 24 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 25 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 26 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 27 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 28 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
[wait-for-migrations] Attempt 29 of 30
[wait-for-migrations] Waiting 30 seconds before next attempt
```

```
kubectl -n awx logs -f deployment/awx-task -c awx-ee
Error: error loading config file: field mintls13 not defined for command tls-client: unknown field name mintls13
```

```
kubectl -n awx logs -f deployment/awx-task -c awx-rsyslog
[wait-for-migrations] Waiting for database migrations...
[wait-for-migrations] Attempt 1 of 30
[wait-for-migrations] Waiting 0.5 seconds before next attempt
[wait-for-migrations] Attempt 2 of 30
[wait-for-migrations] Waiting 1 seconds before next attempt
[wait-for-migrations] Attempt 3 of 30
[wait-for-migrations] Waiting 2 seconds before next attempt
[wait-for-migrations] Attempt 4 of 30
[wait-for-migrations] Waiting 4 seconds before next attempt
[wait-for-migrations] Attempt 5 of 30
[wait-for-migrations] Waiting 8 seconds before next attempt
```

```
kubectl -n awx logs -f deployment/awx-task -c redis
1:C 17 May 2023 18:16:02.666 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 17 May 2023 18:16:02.666 # Redis version=7.0.4, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 17 May 2023 18:16:02.666 # Configuration loaded
1:M 17 May 2023 18:16:02.666 * monotonic clock: POSIX clock_gettime
1:M 17 May 2023 18:16:02.667 * Running mode=standalone, port=0.
1:M 17 May 2023 18:16:02.667 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
1:M 17 May 2023 18:16:02.667 # Server initialized
1:M 17 May 2023 18:16:02.668 * Loading RDB produced by version 7.0.4
1:M 17 May 2023 18:16:02.668 * RDB age 3493331 seconds
1:M 17 May 2023 18:16:02.668 * RDB memory usage when created 1.91 Mb
1:M 17 May 2023 18:16:02.668 * Done loading RDB, keys loaded: 0, keys expired: 21.
1:M 17 May 2023 18:16:02.668 * DB loaded from disk: 0.001 seconds
1:M 17 May 2023 18:16:02.668 * The server is now ready to accept connections at /var/run/redis/redis.sock
1:M 17 May 2023 18:21:03.084 * 100 changes in 300 seconds. Saving...
1:M 17 May 2023 18:21:03.085 * Background saving started by pid 20
20:C 17 May 2023 18:21:03.087 * DB saved on disk
20:C 17 May 2023 18:21:03.087 * Fork CoW for RDB: current 0 MB, peak 0 MB, average 0 MB
1:M 17 May 2023 18:21:03.185 * Background saving terminated with success
1:M 17 May 2023 18:26:04.011 * 100 changes in 300 seconds. Saving...
1:M 17 May 2023 18:26:04.011 * Background saving started by pid 21
21:C 17 May 2023 18:26:04.013 * DB saved on disk
21:C 17 May 2023 18:26:04.013 * Fork CoW for RDB: current 0 MB, peak 0 MB, average 0 MB
1:M 17 May 2023 18:26:04.111 * Background saving terminated with success
```

**Comment by kurokobo** [comment]
@marek1712 
Thanks for updating.

> $ kubectl -n awx logs -f deployment/awx-task -c awx-ee
> Error: error loading config file: field mintls13 not defined for command tls-client: unknown field name mintls13

First of all, this error should be solved. This error means that the configuration file for Receptor that generated by updated AWX Operator can not be handled by Receptor included in the container image for `awx-ee` container.

I see you are using `deployablenl/awx-controlplane-ee:latest` for `awx-ee` container by specifing `control_plane_ee_image`, so I guess the Receptor that included in `deployablenl/awx-controlplane-ee:latest` is outdated and can not handle updated configuration file.

You should sync your container image on your container registry with upstream quay.io, and make K3s to re-pull that image from container registry.

1. Sync your `deployablenl/awx-controlplane-ee:latest` on your container registry with upstream quay.io
2. Remove cached image on K3s: `sudo /usr/local/bin/k3s crictl rmi deployablenl/awx-controlplane-ee:latest`
3. Wait for the next deployment iteration that triggers re-pull that image from your container registry

See if the issue is solved or if another error occurs. For debugging, the commands I provided can be used again.

**Comment by marek1712** [comment]
@kurokobo thank you for the tips.

I used ```deployablenl/awx-controlplane-ee:latest``` because of the cryptography issues in the past. I may as well return to standard awx-ee image.
OK, so I copied your repo again (the one I had on my VM is quite old copy), adjusted towards my config and added ```control_plane_ee_image: quay.io/ansible/awx-ee:latest```
Then as a precaution issued ```sudo /usr/local/bin/k3s crictl rmi deployablenl/awx-controlplane-ee:latest```

Rebooted the VM and:
[log.txt](https://github.com/ansible/awx-operator/files/11505716/log.txt)

```
[root@awx base]# kubectl -n awx logs -f deployment/awx-task -c awx-ee
Error: error loading config file: field mintls13 not defined for command tls-client: unknown field name mintls13
```

EDIT: **my bad!**
Originally I had:
```
  postgres_storage_requirements:
    requests:
      storage: 2Gi
```
while currently it's set to:
```
  postgres_storage_requirements:
    requests:
      storage: 8Gi
```
After changing to 2Gi, update finished and I'm running 22.2.0!
Just one last (lame) question: what'll happen if I'll exceed 2GB of storage for my Postgres DB?

**Comment by kurokobo** [comment]
@marek1712 

> what'll happen if I'll exceed 2GB of storage for my Postgres DB?

Nothing will happen.
The size of `storage` field of the PVs with type `hostPath` is just like _label_ and never be used as actual capacity limitation, so your PVs have no capacity limit.

Refer to the K8s docs:

> There is no limit on how much space an emptyDir or hostPath volume can consume
> https://kubernetes.io/docs/concepts/storage/volumes/#resources

**Comment by kurokobo** [comment]
F.Y.I., I answered this issue here because I couldn't determine if this was an actual bug or a usage issue, but here is a place to manage bugs and RFEs for the developers.
If you have a simple usage question, you should use the mailing list or chat (https://github.com/ansible/awx-operator#get-involved), or it's okay to use the issue page on my repository if your question is related to my guide.

**Comment by marek1712** [comment]
Hi @kurokobo 
Thank you for the help!
I'll definitely use mailing group next time.

Regards,
Marek

**Comment by dominch** [comment]
Hello, 
I just got this particular issue with 2.4.0 and fresh awx installation, everything was ok except awx-ee that has Crashloopbackoff
```
Error: error loading config file: field mintls13 not defined for command tls-client: unknown field name mintls13
```
On my side that was nothing related to postgress volumes, I could not understand why this could solve problem, rather wrong way. The message suggested something about ingress and TLS, I left that with defaults to configure that myself, but because of this error I needed to add those: 
```
spec:
  ingress_type: ingress
  ingress_tls_secret: awx-tls
```
After restarting deployment it worked. Just leaving this here for others.
